buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/gradlebuild/unittestandcompile/UnitTestAndCompilePlugin.kt

```diff
@@ -156,10 +156,8 @@ class UnitTestAndCompilePlugin : Plugin<Project> {
                     it is ExternalDependency || it is FileCollectionDependency
                 }
             )
-        }
-        rootProject.subprojects.forEach { p ->
-            p.plugins.withType<UnitTestAndCompilePlugin> {
-                classpathManifest {
+            rootProject.subprojects.forEach { p ->
+                if(p.plugins.hasPlugin(UnitTestAndCompilePlugin::class.java)) {
```

Comment:
Is this late enough, i.e. will have the subproject already have the plugin applied?

Reply:
I think we cannot make this change in the current setup as the check might be done too early.

I am already working on a PR to remove this configuration code here completely, because it is also very ugly conceptually.

I would suggest to not change anything here in this PR and I'll keep care of getting rid of this cross-project configuration altogether.

It's still good to know how problematic this is!

Reply:
Sounds good to me. Note that there is a similar piece of code in [GradleDistributionPlugin](https://github.com/gradle/gradle/blob/master/buildSrc/subprojects/integration-testing/src/main/kotlin/org/gradle/gradlebuild/test/fixtures/GradleDistributionPlugin.kt#L79) that seems even harder to get rid of.

Reply:
Yes. I am in a battle to get rid of all of this. Hope to have this all removed in a few days. :)